### DISCUSSION TITLE
Does the App have plans to support Bluetooth, album read-write, gyroscope and other capabilities?

### DISCUSSION THREAD

**Discussion by power-hayden** [fyne-io/fyne#3975] [Q&A]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same feature, before opening a new one.
- [X] This issue only relates to a single feature. I will open new issues for any other features.

### Is your feature request related to a problem?

No

### Is it possible to construct a solution with the existing API?

No

### Describe the solution you'd like to see.

I am a novice developer of the Fyne framework. Recently, I was looking at the API documentation and found that the current SDK does not support mobile Bluetooth, album read/write capabilities, etc. Since I hope to use Fyne to develop applications related to Bluetooth and albums, I hope and suggest to support Bluetooth and album read/write capabilities. Is there any plan?

**Comment by Jacalz** [comment]
Support for Bluetooth is in progress by a contributor over at https://github.com/fyne-io/fyne-x/pull/58. I think the plan is to add more specific Android, and mobile features for iOS, like using the camera, reading/writing from/to albums. There might be an open feature request for that somewhere but I’m not sure.

FYI: Please use discussions for questions instead of feature requests. We want to keep the issue tracker for just bugs and feature requests. We also want issues to only contain one feature (or bug) per ticket. I’ll convert this to a discussion.

**Comment by MatejMagat305** [comment]
hi @power-hayden, as @Jacalz said Bluetooth is in early develop, camera maybe will my futures goal, but if you want to add to android own code you can use  	`"fyne.io/fyne/v2/driver"`
```

	driver.RunNative(func(i interface{}) error {
		a, ok := i.(driver.AndroidContext)
		if ok {
			C.some(a.VM, a.Env, a.Ctx)
		}
	})
```
and ask gpt chat for jni ...
